icu4c/source/i18n/unicode/udisplayoptions.h

```diff
@@ -311,6 +311,19 @@ typedef enum UDisplayOptionsSubstituteHandling {
 
 #endif // U_HIDE_DRAFT_API
 
+#ifndef U_HIDE_INTERNAL_API
+
+/**
+ * @return the lowercase CLDR keyword string for the grammatical case and `nullptr` if
+ * the grammatical case is undefined.
+ *
+ * @internal ICU 72
+ */
+U_CAPI const char *U_EXPORT2
+udispopt_getGrammaticalCaseIdentifierOrNull(UDisplayOptionsGrammaticalCase grammaticalCase);
+
+#endif /* U_HIDE_INTERNAL_API */
```

Comment:
Hopefully you can change the code that wants to see null, so that it can accept "undefined" instead. If you do need this kind of helper function, then at least in C it would be better to define it in an internal header file.

Reply:
For now, I am going to just move the check the function to not add extra burden to this PR.